flink-connectors/flink-connector-filesystem/src/main/java/org/apache/flink/streaming/connectors/fs/bucketing/BucketingSink.java

```diff
@@ -722,9 +727,31 @@ public void notifyCheckpointComplete(long checkpointId) throws Exception {
 					}
 				}
 			}
+			isBucketReady(partitionPaths);
 		}
 	}
 
+	@Override
+	public boolean isBucketReady(Set<Path> bucketPathes) {
```

Comment:
seems this implementation of the method has logic problem? shall we `return true;` after iterating the for loop and remove the `return true;` above the `catch`?

Is the purpose of the method for checking all the bucketPathes never end with **pending or inProgress** suffix?

Reply:
If we finish the for loop and haven't returned true yet, then return false. I think this is reasonable. If a file end with pending or inProgress, This means that the file is not ready. and we can not use it.